Add: Connection security level doc (#1247)
home-assistant/companion.home-assistant: docs/getting_started/connection_security_level.md

---
title: "Connection security level"
id: "connection-security-level"
---

> Starting with Companion App version 2025.11, you will have to choose between 2 security options in the app if you want to use a non-encrypted URL to connect to Home Assistant (such as your local IP address for local connection).

Reviewer: We should add a beta flag, set the version for iOS but not android yet.

## Option 1: Most secure

This option makes sure we only connect to a non-encrypted URL (for example, `http://homeassistant.local:8123`) when you are currently on your specified home network.

In the Android app, you can specify your Wi-Fi SSID (for example, "GrandmaHome5G"), VPN, or ethernet access to be considered your home network.

> In iOS, you can specify your Wi-Fi SSID (for example, "GrandmaHome5G") as your home network, but it's not yet possible to use VPN or ethernet in this configuration. If those are required, consider the drawbacks from option 2 and use it if it makes sense.

Author: Updated manually

> On iOS, you can specify your WiFi SSID (for example, "GrandmaHome5G") as your home network, but it's not yet possible (2025.11.0) to use VPN or ethernet in this configuration. If those are required, consider the drawbacks from option 2 and use it if it makes sense.

Reviewer: I would merge this too and add a small note saying that it's not yet available for iOS.

To retrieve the information above, both apps need the location access permission from your device. We understand that this is not desirable for all users, but that's a restriction imposed by Apple platforms and Android. This also doesn't mean that your location will be used for anything besides retrieving the network information. You are always in control if you want to share your location to your local Home Assistant server. Just access **Companion App Settings** > **Sensors** to define what to share and what not to share.

:::warning Important

> To work properly in the background, the location permission needs to also be set to allow **Always** and **Precise**.

Reviewer: ```suggestion
The location permission must be set to **Always** and **Precise** for the app to function correctly in the background.
```

:::

## Option 2: Less secure

> In case you prefer to not allow the app to know your location but still want to access a non-encrypted URL (for example, `http://homeassistant.local:8123`), you can choose this option, which will allow its usage. Keep in mind that this is not recommended at all, since it can expose your Home Assistant access to someone observing network traffic on public WiFi networks, for example.

Reviewer: At the end or at the start we should add a section, explaining the reasoning behind these options. Like why there is a risk in the first place, we could add an advance section explaining the attack vector.

It should be written that is it to protect your data and align with our core values that the user should remain the one that choose.

---
title: "Connection security level"
id: "connection-security-level"
---
![iOS](/assets/iOS.svg) <span class='beta'>BETA 2025.11.0</span>

> Starting with Companion App version 2025.11, you will have to choose between 2 security options in the app if you want to use a non-encrypted URL to connect to Home Assistant (such as your local IP address for local connection).

Reviewer 2: non-encrypted > unencrypted

> Starting with Companion App version 2025.11, you will have to choose between 2 security options in the app if you want to use a non-encrypted URL to connect to Home Assistant (such as your local IP address for local connection).

Reviewer 2: 2 security options > two security options

([spell out numerals below 10 in body text](https://learn.microsoft.com/en-us/style-guide/numbers#numerals-vs-words))

You can specify your WiFi SSID (for example, "GrandmaHome5G"), VPN, or Ethernet access to be considered your home network.

:::info ![iOS](/assets/iOS.svg)
<p>Currently VPN and Ethernet home network configurations are only available on Android.</p>
:::

To retrieve the information above, both apps need the location access permission from your device, since network information is considered one factor for location positioning. We understand that this is not desirable for all users, but that's a restriction imposed by Apple platforms and Android. This means that your location is never going to be used for anything besides retrieving the network information. You are always in control if you want to share your location to your local Home Assistant server.

:::warning Important
The location permission must be set to **Always** and **Precise** for the app to function correctly in the background.
:::

## Option 2: Less secure

> If you don’t grant location permission or disable location at the OS level but still need to access a non-encrypted URL (e.g., http://homeassistant.local:8123), you can choose this option. Use it only if absolutely necessary; **it is not recommended**, as it may expose your Home Assistant data to network eavesdropping, especially on public Wi-Fi.

Reviewer 2: Consistent usage of examples

```suggestion
If you don’t grant location permission or disable location at the OS level but still need to access a non-encrypted URL (for example, http://homeassistant.local:8123), you can choose this option. Use it only if absolutely necessary; **it is not recommended**, as it may expose your Home Assistant data to network eavesdropping, especially on public Wi-Fi.
```
